lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -967,17 +1054,31 @@ def alerts(self) -> dict:
 
             try:
                 scrape_metadata = json.loads(relation.data[relation.app]["scrape_metadata"])
-                topology = JujuTopology.from_relation_data(scrape_metadata)
-
-                alerts[topology.identifier] = alert_rules
-
+                identifier = ProviderTopology.from_relation_data(scrape_metadata).identifier
+                alerts[identifier] = alert_rules
             except KeyError as e:
-                logger.error(
-                    "Relation %s has invalid data : %s",
+                logger.warning(
+                    "Relation %s has no 'scrape_metadata': %s",
                     relation.id,
                     e,
                 )
 
+                if "groups" not in alert_rules:
+                    logger.warning("No alert groups were found in relation data")
+                    continue
+                # Construct an ID based on what's in the alert rules
+                for group in alert_rules["groups"]:
+                    try:
+                        labels = group["rules"][0]["labels"]
```

Comment:
@simskij @mmanciop @sed-i @rbarry82 
Is it safe to assume that all rules in a group will have the same topology labels ? Note this is across the entire ecosystem of observability charms - cos config, lma (cos) proxy, scrape config etc.

Reply:
iiuc this is the rules provider side so yes, all alerts will have the same topology (unless grafana agent intervenes and applies this change to rules that are not its own..?)